src/sentry/incidents/logic.py

```diff
@@ -263,47 +265,70 @@ def create_event_stat_snapshot(incident, start, end):
 
 
 def build_incident_query_params(incident, start=None, end=None):
-    params = {
-        'start': incident.date_started if start is None else start,
-        'end': incident.current_end_date if end is None else end,
-    }
-    group_ids = list(IncidentGroup.objects.filter(
-        incident=incident,
-    ).values_list('group_id', flat=True))
-    if group_ids:
-        params['issue.id'] = group_ids
-    project_ids = list(IncidentProject.objects.filter(
-        incident=incident,
-    ).values_list('project_id', flat=True))
-    if project_ids:
-        params['project_id'] = project_ids
+    return bulk_build_incident_query_params([incident], start=start, end=end)[0]
+
+
+def bulk_build_incident_query_params(incidents, start=None, end=None):
+    incident_groups = defaultdict(list)
+    for incident_id, group_id in IncidentGroup.objects.filter(
+        incident__in=incidents,
+    ).values_list('incident_id', 'group_id'):
+        incident_groups[incident_id].append(group_id)
+    incident_projects = defaultdict(list)
+    for incident_id, project_id in IncidentProject.objects.filter(
+        incident__in=incidents,
+    ).values_list('incident_id', 'project_id'):
+        incident_projects[incident_id].append(project_id)
+
+    query_args_list = []
+    for incident in incidents:
+        params = {
+            'start': incident.date_started if start is None else start,
+            'end': incident.current_end_date if end is None else end,
+        }
+        group_ids = incident_groups[incident.id]
+        if group_ids:
+            params['issue.id'] = group_ids
+        project_ids = incident_projects[incident.id]
+        if project_ids:
+            params['project_id'] = project_ids
+        query_args_list.append(get_snuba_query_args(incident.query, params))
 
-    return get_snuba_query_args(incident.query, params)
+    return query_args_list
 
 
 def get_incident_event_stats(incident, start=None, end=None, data_points=50):
     """
     Gets event stats for an incident. If start/end are provided, uses that time
     period, otherwise uses the incident start/current_end.
     """
-    kwargs = build_incident_query_params(incident, start=start, end=end)
-    rollup = max(int(incident.duration.total_seconds() / data_points), 1)
-    return SnubaTSResult(
-        raw_query(
+    query_params = build_incident_query_params(incident, start=start, end=end)
```

Comment:
This method calls the `bulk_build_incident_query_params` function and pops off the first element which you then re-wrap in another list. Would calling `bulk_build_incident_query_params` directly be simpler?

Reply:
Yeah that makes sense, not a lot of point using a non-bulk method to get things to pass to a bulk method.